backbone.js

```diff
@@ -112,42 +101,46 @@
 
       // Remove all callbacks for all events.
       if (!name && !callback && !context) {
-        this._events = void 0;
+        var listenerIds = _.keys(this._listeners);
+        for (var i = 0, l = listenerIds.length; i < l; i++) {
```

Comment:
On a related note... Should we be avoiding `_.keys` in `_.each`? What if we branched at `nativeKeys` vs `for ... in` in `each` to avoid the double loop that @stephanebachelier pointed out? @jdalton